### PR TITLE
Moving QueueHealth and SubscriptionHealth to use PeekMessagesAsync

### PR DESCRIPTION
**Pull request by Tiberriver256**
**What this PR does / why we need it**:

Moves QueueHealth and SubscriptionHealth to using PeekMessagesAsync

**Which issue(s) this PR fixes**:

Please reference the issue this PR will close: #1503

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [ ] Code compiles correctly
- [ ] Created/updated tests
- [ ] Unit tests passing
- [ ] End-to-end tests passing
- [ ] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs [hunk 1]
```diff
@@ -33,8 +33,9 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         {
             try
             {
-                var managementClient = ManagementClientConnections.GetOrAdd(ConnectionKey, _ => CreateManagementClient());
-                _ = await managementClient.GetQueueRuntimePropertiesAsync(_queueName, cancellationToken);
+                await using var client = ClientConnections.GetOrAdd(ConnectionKey, _ => CreateClient());
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs:36]: `using` here 😕 ? Clients from dictionary are long-lived

File: src/HealthChecks.AzureServiceBus/AzureServiceBusSubscriptionHealthCheck.cs [hunk 2]
```diff
@@ -45,8 +45,9 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         {
             try
             {
-                var managementClient = ManagementClientConnections.GetOrAdd(ConnectionKey, _ => CreateManagementClient());
-                _ = await managementClient.GetSubscriptionRuntimePropertiesAsync(_topicName, _subscriptionName, cancellationToken);
+                await using var client = ClientConnections.GetOrAdd(ConnectionKey, _ => CreateClient());
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusSubscriptionHealthCheck.cs:48]: the same here

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs (hunk 1 above)
Comment (Tiberriver256) [reply src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs]: Sorry about that. I think making the AzureServiceBusHealthCheck class implement IAsyncDisposable is the right move but not 100%. Let me know if that looks good.

Comment (Tiberriver256) [reply src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs]: Totally just realized I was dealing with a static property so I'm guessing they should actually just never be disposed.

Comment (sungam3r) [reply src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs]: Yes

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs [hunk 3]
```diff
@@ -33,8 +33,9 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         {
             try
             {
-                var managementClient = ManagementClientConnections.GetOrAdd(ConnectionKey, _ => CreateManagementClient());
-                _ = await managementClient.GetQueueRuntimePropertiesAsync(_queueName, cancellationToken);
+                var client = ClientConnections.GetOrAdd(ConnectionKey, _ => CreateClient());
+                var receiver = client.CreateReceiver(_queueName);
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs:37]: `CreateReceiver` returns something that doesn't implement `IDisposable`. Right?

File: test/HealthChecks.AzureServiceBus.Tests/HealthChecks.AzureServiceBus.approved.txt [hunk 4]
```diff
@@ -6,14 +6,17 @@ namespace HealthChecks.AzureServiceBus
         public AzureEventHubHealthCheck(string connectionString, string eventHubName) { }
         public System.Threading.Tasks.Task<Microsoft.Extensions.Diagnostics.HealthChecks.HealthCheckResult> CheckHealthAsync(Microsoft.Extensions.Diagnostics.HealthChecks.HealthCheckContext context, System.Threading.CancellationToken cancellationToken = default) { }
     }
-    public abstract class AzureServiceBusHealthCheck
+    public abstract class AzureServiceBusHealthCheck : System.IAsyncDisposable
```
Comment (sungam3r) [inline test/HealthChecks.AzureServiceBus.Tests/HealthChecks.AzureServiceBus.approved.txt:9]: oops

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs (hunk 3 above)
Comment (Tiberriver256) [reply src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs:37]: Good catch. Looks like it actually implements IAsyncDisposable and is recommended to be cached.

From the comments on `ServiceBusReceiver`:

>  /// The <see cref="ServiceBusReceiver" /> is safe to cache and use for the lifetime of an
    /// application or until the <see cref="ServiceBusClient" /> that it was created by is disposed.
    /// Caching the receiver is recommended when the application is consuming messages
    /// regularly or semi-regularly.

I'll refactor to be caching that in addition to the ServiceBusClient

File: src/HealthChecks.AzureServiceBus/AzureServiceBusSubscriptionHealthCheck.cs [hunk 5]
```diff
@@ -45,8 +45,11 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         {
             try
             {
-                var managementClient = ManagementClientConnections.GetOrAdd(ConnectionKey, _ => CreateManagementClient());
-                _ = await managementClient.GetSubscriptionRuntimePropertiesAsync(_topicName, _subscriptionName, cancellationToken);
+                var client = ClientConnections.GetOrAdd(ConnectionKey, _ => CreateClient());
+                var receiver = ServiceBusReceivers
+                    .GetOrAdd($"{ConnectionKey}-{_topicName}-{_subscriptionName}", client.CreateReceiver(_topicName, _subscriptionName));
```
[the remark this answers is not in the document]
Comment (Tiberriver256) [reply src/HealthChecks.AzureServiceBus/AzureServiceBusSubscriptionHealthCheck.cs:50]: Ooooo... you catch all the good ones sir

File: src/HealthChecks.AzureServiceBus/AzureServiceBusSubscriptionHealthCheck.cs [hunk 6]
```diff
@@ -45,8 +45,11 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         {
             try
             {
-                var managementClient = ManagementClientConnections.GetOrAdd(ConnectionKey, _ => CreateManagementClient());
-                _ = await managementClient.GetSubscriptionRuntimePropertiesAsync(_topicName, _subscriptionName, cancellationToken);
+                var client = ClientConnections.GetOrAdd(ConnectionKey, _ => CreateClient());
+                var receiver = ServiceBusReceivers
+                    .GetOrAdd($"{nameof(AzureServiceBusSubscriptionHealthCheck)}_{ConnectionKey}", client.CreateReceiver(_topicName, _subscriptionName));
+
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusSubscriptionHealthCheck.cs:51]:
```suggestion
                var receiver = ServiceBusReceivers.GetOrAdd($"{nameof(AzureServiceBusSubscriptionHealthCheck)}_{ConnectionKey}", client.CreateReceiver(_topicName, _subscriptionName));
```

**Review by sungam3r — APPROVED** [review]

**Pull request merged**